coderd/database/pubsub/pubsub.go

```diff
@@ -352,11 +353,61 @@ func (p *PGPubsub) recordReconnect() {
 	}
 }
 
+// logDialer is a pq.Dialer and pq.DialerContext that logs when it starts
+// connecting and when the TCP connection is established.
+type logDialer struct {
+	logCtx context.Context
```

Comment:
I don't think we need this, just use `context.Background()` for logging.  We don't squirrel away any extra slog fields on the context anywhere I'm aware of.